web-registry/src/components/organisms/BuyModalOptions/BuyModalOptions.tsx

```diff
@@ -54,11 +62,28 @@ export const BuyModalOptions = ({
               button={{
                 text: card?.button?.buttonText ?? '',
                 onClick: card?.button?.buttonLink
-                  ? () => onBtnClick(openModal, card?.button)
-                  : () =>
+                  ? () => {
+                      if (track) {
+                        track('buyScheduleCall', {
+                          url: location.pathname,
+                          projectId: selectedProject?.id,
+                          creditClassId: selectedProject?.creditClassId,
+                        });
+                      }
+                      onBtnClick(openModal, card?.button);
+                    }
+                  : () => {
+                      if (track) {
+                        track('buyKeplr', {
+                          url: location.pathname,
+                          projectId: selectedProject?.id,
+                          creditClassId: selectedProject?.creditClassId,
+                        });
+                      }
```

Comment:
TBH i'd prefer to leave this as is, this is readable and easy to understand IMO.
I should probably add the typing info here though.

Reply:
TBH i still don't feel too strongly about it, but i think your opinion is meaningful here, so i'm glad to do this. one question i have is, is there a strong reason to use a hook here rather than a regular function?

Reply:
Gave this a go here: 07e2bdd

Reply:
Thanks for your changes. Regarding hooks vs functions, if a hook is not using any react primitive then it could definitely be a function. In that case, these 2 tracking functions should probably be just function yes. Using plain function also helps to reduce rerenders since function ref won't change, it can be achieved in hooks by using `useCallback` but it's more verbose. 
Feel free to change it or not for this PR it's not a big deal here since there are no child components.